src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-empty.c

```diff
@@ -0,0 +1,70 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <errno.h>
+#include <libcryptsetup.h>
+
+#include "cryptsetup-token.h"
+#include "macro.h"
+#include "version.h"
+
+#define TOKEN_NAME "systemd-empty"
+#define TOKEN_VERSION_MAJOR "1"
+#define TOKEN_VERSION_MINOR "0"
+
+/* for libcryptsetup debug purpose */
+_public_ const char *cryptsetup_token_version(void) {
+
+        return TOKEN_VERSION_MAJOR "." TOKEN_VERSION_MINOR " systemd-v" STRINGIFY(PROJECT_VERSION) " (" GIT_VERSION ")";
+}
+
+_public_ int cryptsetup_token_open_pin(
+                struct crypt_device *cd, /* is always LUKS2 context */
+                int token /* is always >= 0 */,
+                const char *pin,
+                size_t pin_size,
+                char **ret_password, /* freed by cryptsetup_token_buffer_free */
+                size_t *ret_password_len,
+                void *usrptr /* plugin defined parameter passed to crypt_activate_by_token*() API */) {
+
+        assert(token >= 0);
+        assert(!pin || pin_size > 0);
+        assert(ret_password);
+        assert(ret_password_len);
+
+        /* free'd automatically by libcryptsetup */
+        *ret_password = strdup("");
```

Comment:
strdup() might fail. And we have this rule that we shold never touch return parameters on failure. hence use temporary variable, i.e.

```c
_cleanup_free_ char *p = NULL;

p = strdup("");
if (!p) 
        return crypt_log_oom(cd);

*ret_password = TAKE_PTR(p);
*ret_password_len = 0;
return 0;
```